packages/zk-eddsa-ticket-pcd/test/ZKEdDSATicketPCD.spec.ts

```diff
@@ -0,0 +1,218 @@
+import { ArgumentTypeName } from "@pcd/pcd-types";
+import {
+  SemaphoreIdentityPCDPackage,
+  SemaphoreIdentityPCDTypeName
+} from "@pcd/semaphore-identity-pcd";
+// eslint-disable-next-line @typescript-eslint/triple-slash-reference
+/// <reference path="../src/util/declarations/circomlibjs.d.ts" />
+/// <reference path="../src/util/declarations/snarkjs.d.ts" />
+import { Identity } from "@semaphore-protocol/identity";
+import { expect } from "chai";
+import "mocha";
+import * as path from "path";
+import { v4 as uuid } from "uuid";
+import {
+  EdDSATicketFieldsRequest,
+  ZKEdDSATicketPCD,
+  ZKEdDSATicketPCDArgs,
+  ZKEdDSATicketPCDPackage,
+  ZKEdDSATicketPCDTypeName
+} from "../src";
+
+const zkeyFilePath = path.join(
+  __dirname,
+  `../artifacts-unsafe/eddsaTicket.zkey`
+);
+const wasmFilePath = path.join(
+  __dirname,
+  `../artifacts-unsafe/eddsaTicket_js/eddsaTicket.wasm`
+);
+
+let toArgs: (
+  ticketData: ITicketData,
+  fieldsRequested: EdDSATicketFieldsRequest,
+  withNullifier: boolean
+) => Promise<ZKEdDSATicketPCDArgs>;
+
+const identity = new Identity(
+  '["329061722381819402313027227353491409557029289040211387019699013780657641967", "99353161014976810914716773124042455250852206298527174581112949561812190422"]'
+);
+
+// Key borrowed from https://github.com/iden3/circomlibjs/blob/4f094c5be05c1f0210924a3ab204d8fd8da69f49/test/eddsa.js#L103
+const prvKey =
+  "0001020304050607080900010203040506070809000102030405060708090001";
+
+const WATERMARK = BigInt(6);
+const EXTERNAL_NULLIFIER = BigInt(42);
+
+describe("EdDSA partial ticket should work", function () {
+  this.timeout(1000 * 30);
+
+  let pcd1: ZKEdDSATicketPCD;
+
+  const ticketData1: ITicketData = {
+    // the fields below are not signed and are used for display purposes
+
+    attendeeName: "test name",
+    attendeeEmail: "user@test.com",
+    eventName: "event",
+    ticketName: "ticket",
+    checkerEmail: "checker@test.com",
+
+    // the fields below are signed using the server's private eddsa key
+
+    ticketId: "b38501b0-cf66-4416-a4ef-631a9a9b32c4",
+    eventId: "4332f16c-8444-4261-94b9-a0ba8ca917e2",
+    productId: "c94002fb-2c41-480b-842d-fa826fb517e5",
+    timestampConsumed: 0,
+    timestampSigned: 1693028498280,
+    attendeeSemaphoreId: identity.getCommitment().toString(),
+    isConsumed: false,
+    isRevoked: false
+  };
+
+  const fieldsRequested1: EdDSATicketFieldsRequest = {
+    revealTicketId: false,
+    revealEventId: true,
+    revealProductId: true,
+    revealTimestampConsumed: false,
+    revealTimestampSigned: false,
+    revealAttendeeSemaphoreId: false,
+    revealIsConsumed: false,
+    revealIsRevoked: false
+  };
+
+  this.beforeAll(async function () {
+    await EdDSAPCDPackage.init!({});
+    await EdDSATicketPCDPackage.init!({});
+    if (!ZKEdDSATicketPCDPackage.init) return;
+    await ZKEdDSATicketPCDPackage.init({
+      zkeyFilePath,
+      wasmFilePath
+    });
+
+    toArgs = async (
```

Comment:
move this function definition to the declaration of `toArgs`?

Reply:
can't unfortunately - requires packages to have been `init`ed. unless i can do this outside of the `beforeAll`?